cli/src/main.rs

```diff
@@ -43,8 +46,15 @@ impl FromStr for Alphabet {
                 if bytes.len() != 58 {
                     return Err(anyhow!("custom alphabet is not 58 characters long"));
                 }
-                let ptr = bytes.as_ptr() as *const [u8; 58];
-                Alphabet::Custom(unsafe { *ptr })
+                // SAFETY: Length checked just above
+                let bytes = unsafe { &*(bytes.as_ptr().cast::<[u8; 58]>()) };
```

Comment:
This is just `transmute`, right? Better to go through transmute so you get the runtime checks that this is valid.

Reply:
No, it's going from a `&[u8]` thicc reference to a `&[u8; 58]` thin reference, via getting the `*const u8` pointer to the start of the slice and widening that to a `*const [u8; 58]` pointer to the 58 bytes needed.

Reply:
(Though, if I just wait 2 days I can replace it with the `TryFrom` impl below when 1.47 releases 😁)